pengine/graph.c

```diff
@@ -1079,6 +1098,10 @@ action2xml(action_t * action, gboolean as_input, pe_working_set_t *data_set)
         add_downed_nodes(action_xml, action, data_set);
     }
 
+    if (safe_str_eq(action->task, CRM_OP_MAINTENANCE_NODES)) {
+        add_maintenance_nodes(action_xml, action, data_set);
+    }
+
```

Comment:
I'm a little concerned about all the times we loop through resources, nodes, and actions already. I think as we start looking how to improve performance at scale, that will be one of the bottlenecks we'll need to take care of.

So, let's see if we can do this in one of the existing node loops in stage2(), stage5() or stage6(). Maintenance mode should be finally known already after stage0() when the cluster status is unpacked.